Frontend/src/app/insert-match/insert-match.component.ts

```diff
@@ -95,16 +102,31 @@ export class InsertMatchComponent implements OnInit, OnChanges {
 		return true;
 	}
 
+	checkFinals(finals){
+		if(finals.length > 0)
+			this.hasFinal = true;
+		else
+			this.hasFinal = false;
+	}
 	tryCreateMatch() {
 		this.match.isFinal = this.isFinal;
 
+		// TODO: get championship of month:
+		this.match.championship = this.championshipService.getCurrentChampionship();
+
+		this.matchService.getFinalFromChampionship(this.match.championship._id).subscribe(
+			(res) => {
+				this.checkFinals(res);
+			},
+			(error : any) => {
+				console.log(error);
+				this.error = error;
+			});
+
 		if (!this.validate()) {
```

Comment:
Yeah, I thought about that. I'm also getting some errors from this PR while saving matches. It seems like the id from the current championship takes a while to be set as well, and it's not always rendered.